tracer.go

```diff
@@ -1319,6 +1341,31 @@ func (t *Tracer) gatherMetrics(ctx context.Context, gatherers []MetricsGatherer,
 	}()
 }
 
+// canDropUnsampledTx returns true when the remote APM Server's major version is
+// >= 8, otherwise returns false.
+func (t *Tracer) canDropUnsampledTx(ctx context.Context) bool {
+	if t.versionGetter == nil {
+		return false
+	}
+
+	// Cancel the operation after 5s, we do not want to block the loop for more
+	// than necessary.
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+	if v, err := t.versionGetter.GetVersion(ctx); err == nil {
+		return parseMajorVersion(v) >= 8
+	}
+	return false
+}
+
+func (t *Tracer) dropUnsampledTx(td *TransactionData, sampled, dropUnsampled bool) bool {
+	if !sampled && dropUnsampled {
+		td.reset(t)
+		return true
+	}
+	return false
+}
```

Comment:
I also wonder if we should avoid blocking here, and only drop non-sampled transactions if the version has already been fetched? That might mean sending a few non-sampled transactions to the server while the agent is starting up, but I think that's probably a reasonable tradeoff, to avoid blocking other events.

Reply:
Sounds good, I think we can just cache the parsed major version as an integer in the HTTPTransport layer. Additionally, we can set the deadline to `1us` so the context is cancelled if the version isn't cached.